### ISSUE TITLE
Unread image is not labelled

### ISSUE THREAD

**Issue by alexarnaud** [lwindolf/liferea#715]
Hello,

Environment:
- Debian Testing
- Liferea 1.12.6
- Latest Orca screen reader version

I'm visual-impaired and I'm running the Orca screen reader to use  Liferead only with keyboard shortcuts.

Unlabelled image are announce by the screen reader as "image" whereas it should be useful to have "unread" on it to know what it is.

Best regards,
Alex.

**Comment by alexarnaud** [comment]
I've updated the issue because it's a more general issue. It looks like all the icons are impacted by this.

I'm trying to find a way to figure out this but as it looks like it's not only managed by glade and at this time for me it's like a labyrinth.

Best regard,
Alex.

**Comment by alexarnaud** [comment]
OK, images are renderer with GtkCellRendererPixbuf, see the documentation here: https://developer.gnome.org/gtk3/stable/GtkCellRendererPixbuf.html

I don't find a way to add a label with a Pixbuf. Am I so beginner or is it really not permitted by this renderer ?

Best regards,
Alex.

**Comment by Leiaz** [comment]
The relevant code is in `src/ui/icons.c' and `src/ui/item_list_view.c`, the cell renderer for this column is created at line 811. We set the "gicon" property of the renderer. The `GIcon` used for the item state is a `GThemedIcon`, so it is an icon name. Gtk could use that name as default text. I don't know if we have to directly use ATK : https://wiki.gnome.org/Accessibility/Documentation/GNOME2/AtkGuide/Gtk That guide looks a bit outdated.

**Comment by alexarnaud** [comment]
> The relevant code is in `src/ui/icons.c` and `src/ui/item_list_view.c` , the cell renderer for this column is created at line 811. We set the "gicon" property of the renderer. The `GIcon` used for the item state is a `GThemedIcon`, so it is an icon name.

Thanks for the pointer.

> Gtk could use that name as default text. I don't know if we have to directly use ATK : https://wiki.gnome.org/Accessibility/Documentation/GNOME2/AtkGuide/Gtk That guide looks a bit outdated.

At this time, I don't know. I've ask that on GNOME Accessibility Devel mailling list to have the opinion of a11y expert about that.

Best regards,
Alex.

**Comment by lwindolf** [comment]
@alexarnaud as an Orca user I could imagine that you are not interested at all at those icons, could this be? Because those GtkCellRenderer only shows the news feed favicons (or a default icon if the feed doesn't have one). So it is more a visual signal to make feed identification easier. This signal might be uninteresting for non-visual users.

So it might be easier to hide this renderer from screen-reader use cases.

What do you think?

**Comment by alexarnaud** [comment]
> @alexarnaud as an Orca user I could imagine that you are not interested at all at those icons, could this be? Because those GtkCellRenderer only shows the news feed favicons (or a default icon if the feed doesn't have one). So it is more a visual signal to make feed identification easier. This signal might be uninteresting for non-visual users.

You're correct about the favicon but we'd like to be able to know the read/unread state.

Best regards,
Alex.

**Comment by alexarnaud** [comment]
It looks like it's not possible to add a label to an image with the Pixbuf renderer, see https://discourse.gnome.org/t/gtkcellrendererpixbuf-doesnt-allow-to-add-a-label-for-image/699/2

Best regards,
Alex.

**Comment by alexarnaud** [comment]
Someone point me to this: https://stackoverflow.com/questions/14223296/with-gtk3-how-do-i-make-a-tree-view-accessible

Best regards,
Alex.

**Comment by pvagner** [comment]
There is almost no documentation but roughly speaking you will need:
* Look into the GTK source code for GtkRendererCellAccessible and subclass it. gtktextcellaccessible might be good to look at too if you would like to get some inspiration.
* For the read status icon I think you should at least implement AtkObject.get_name and update_cache methods.
* Then use the gtk_cell_renderer_class_set_accessible_type method on your cell renderer object to register your new subclass https://developer.gnome.org/gtk3/stable/GtkCellRenderer.html#gtk-cell-renderer-class-set-accessible-type

Read this email list posting to get the analogy to the gtktextcellaccessible explained: https://mail.gnome.org/archives/gnome-accessibility-list/2014-July/msg00014.html or feel free to read the whole email list thread although it contains a lot of wala specific guess work.

**Comment by Leiaz** [comment]
The issue #643 about favicons and feed names is related. We could also have an option for each of those columns, to have them display text or an icon.

**Comment by pvagner** [comment]
Eventhough you will render text instead of images on those cell renderers it won't make them accessible I guess. So custom cell renderer accessible might still be usefull.

**Comment by lwindolf** [comment]
Closing this. There are currently no plans to add additional features.
